src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java

```diff
@@ -0,0 +1,118 @@
+import com.researchspace.dcd.utils.DigitalCommonsDataAdapterUtils;
+import com.researchspace.repository.spi.IDepositor;
+import com.researchspace.repository.spi.IRepository;
+import com.researchspace.repository.spi.LicenseConfigInfo;
+import com.researchspace.repository.spi.RepositoryConfig;
+import com.researchspace.repository.spi.RepositoryConfigurer;
+import com.researchspace.repository.spi.RepositoryOperationResult;
+import com.researchspace.repository.spi.Subject;
+import com.researchspace.repository.spi.SubmissionMetadata;
+import com.researchspace.repository.spi.properties.RepoProperty;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.client.RestClientException;
+
+
+@Getter
+@Setter
+@Slf4j
+public class DigitalCommonsDataRepository implements IRepository, RepositoryConfigurer {
+
+  private DigitalCommonsDataClient digitalCommonsDataClient;
+
+  @Override
+  public void configure(RepositoryConfig repositoryConfig) {
+    this.digitalCommonsDataClient = new DigitalCommonsDataClientImpl(
+        repositoryConfig.getServerURL(),
+        repositoryConfig.getIdentifier());
+  }
+
+  @Override
+  public RepositoryOperationResult submitDeposit(IDepositor iDepositor, File file,
+      SubmissionMetadata submissionMetadata, RepositoryConfig repositoryConfig) {
+    DigitalCommonDataSubmission datasetReequest = createDatasetRequest(submissionMetadata);
+
+    try {
+
+      DigitalCommonsDataDataset deposition = digitalCommonsDataClient.createDataset(
+          datasetReequest);
+      DigitalCommonsDataFile depositedFile = digitalCommonsDataClient.depositFile(deposition,
+          file.getName(), file);
+      return new RepositoryOperationResult(true,
+          "Export uploaded to DigitalCommonsData successfully.",
+          new URL(repositoryConfig.getServerURL() + "/drafts/" + deposition.getId()));
+
+    } catch (RestClientException e) {
+      log.error("RestClientException occurred while submitting to DigitalCommonsData", e);
+      return new RepositoryOperationResult(false,
+          "RestClientException occurred while submitting to DigitalCommonsData", null);
+    } catch (MalformedURLException e) {
+      log.error("MalformedURLException occurred while submitting to DigitalCommonsData", e);
+      return new RepositoryOperationResult(false,
+          "MalformedURLException occurred while submitting to DigitalCommonsData", null);
+    } catch (IOException e) {
+      log.error("IOException occurred while submitting to DigitalCommonsData", e);
+      return new RepositoryOperationResult(false,
+          "IOException occurred while submitting to DigitalCommonsData", null);
+    }
```

Comment:
a lot of repeated code - would be more readable with a `catch-multiple` syntax

Reply:
grouped now